web3/utils/events.py

```diff
@@ -210,7 +213,7 @@ def get_event_data(event_abi, log_entry):
     ))
 
     event_data = {
-        'args': event_args,
+        'args': AttributeDict(event_args),
```

Comment:
Could do an `AttributeDict.recursive(event_args)` here (or even to all of the `event_data`) and skip the one in web3/contract.

Up to you